### ISSUE TITLE
rootless podman cgroupfs error

### ISSUE THREAD

**Issue by lfarkas** [containers/libpod#3001]
I try to run a minimal docker container with rootless podman. I assume official mariadb is a simple default test. 

it's the latest fedora 29 and selinux disabled with:
podman-1.2.0-2.git3bd528e.fc29.x86_64
libgpod-0.8.3-25.fc29.x86_64
runc-1.0.0-85.dev.gitdd22a84.fc29.x86_64
kernel-5.0.8-200.fc29.x86_64

isn't such a basic service should have to run with podman?

first with a lots of option but finally only a few:
podman run --log-level=debug --rm -ti --env-file .env -v /srv/vidux/mariadb/db:/var/lib/mysql mariadb:1

it's always fail with:
```
DEBU[0000] running conmon: /usr/libexec/podman/conmon    args=[-c 53ef933d3f6720e5a250257a742bd758e71795077c519acb782ee49162d4be34 -u 53ef933d3f6720e5a250257a742bd758e71795077c519acb782ee49162d4be34 -r /usr/bin/runc -b /home/lfarkas/.local/share/containers/storage/overlay-containers/53ef933d3f6720e5a250257a742bd758e71795077c519acb782ee49162d4be34/userdata -p /tmp/1000/overlay-containers/53ef933d3f6720e5a250257a742bd758e71795077c519acb782ee49162d4be34/userdata/pidfile -l /home/lfarkas/.local/share/containers/storage/overlay-containers/53ef933d3f6720e5a250257a742bd758e71795077c519acb782ee49162d4be34/userdata/ctr.log --exit-dir /run/user/1000/libpod/tmp/exits --conmon-pidfile /home/lfarkas/.local/share/containers/storage/overlay-containers/53ef933d3f6720e5a250257a742bd758e71795077c519acb782ee49162d4be34/userdata/conmon.pid --exit-command /usr/bin/podman --exit-command-arg --root --exit-command-arg /home/lfarkas/.local/share/containers/storage --exit-command-arg --runroot --exit-command-arg /tmp/1000 --exit-command-arg --log-level --exit-command-arg debug --exit-command-arg --cgroup-manager --exit-command-arg cgroupfs --exit-command-arg --tmpdir --exit-command-arg /run/user/1000/libpod/tmp --exit-command-arg --runtime --exit-command-arg runc --exit-command-arg --storage-driver --exit-command-arg overlay --exit-command-arg container --exit-command-arg cleanup --exit-command-arg --rm --exit-command-arg 53ef933d3f6720e5a250257a742bd758e71795077c519acb782ee49162d4be34 --socket-dir-path /run/user/1000/libpod/tmp/socket -t --log-level debug --syslog]
WARN[0000] Failed to add conmon to cgroupfs sandbox cgroup: mkdir /sys/fs/cgroup/systemd/libpod_parent: permission denied 
DEBU[0000] Cleaning up container 53ef933d3f6720e5a250257a742bd758e71795077c519acb782ee49162d4be34 
DEBU[0000] Network is already cleaned up, skipping...   
DEBU[0000] unmounted container "53ef933d3f6720e5a250257a742bd758e71795077c519acb782ee49162d4be34" 
DEBU[0000] Cleaning up container 53ef933d3f6720e5a250257a742bd758e71795077c519acb782ee49162d4be34 
DEBU[0000] Network is already cleaned up, skipping...   
DEBU[0000] Storage is already unmounted, skipping...    
DEBU[0000] Storage is already unmounted, skipping...    
ERRO[0000] error reading container (probably exited) json message: EOF 
```
or without -ti:
```
WARN[0000] Failed to add conmon to cgroupfs sandbox cgroup: mkdir /sys/fs/cgroup/systemd/libpod_parent: permission denied 
DEBU[0000] Received container pid: -1                   
DEBU[0000] Cleaning up container 03e96af941652af9b93102e7121ade28c2d6f4a97072081fc4a1570f47f88eb4 
DEBU[0000] Network is already cleaned up, skipping...   
DEBU[0000] unmounted container "03e96af941652af9b93102e7121ade28c2d6f4a97072081fc4a1570f47f88eb4" 
DEBU[0000] Cleaning up container 03e96af941652af9b93102e7121ade28c2d6f4a97072081fc4a1570f47f88eb4 
DEBU[0000] Network is already cleaned up, skipping...   
DEBU[0000] Storage is already unmounted, skipping...    
DEBU[0000] Storage is already unmounted, skipping...    
ERRO[0000] container create failed: container_linux.go:345: starting container process caused "process_linux.go:424: container init caused \"write /proc/self/attr/keycreate: invalid argument\""
: internal libpod error 
```

**Comment by mheon** [comment]
This is an SELinux-related error coming out of `runc`. To you have SELinux enabled or disabled? Can you get the output of `rpm -qa container-selinux`?

**Comment by lfarkas** [comment]
container-selinux-2.95-1.gite3ebc68.fc29.noarch

**Comment by lfarkas** [comment]
it's working if i set selinux permissive. IMHO it's nonsense that anything is required any feature which is not essential! why do you depend on selinux at all?

**Comment by mheon** [comment]
SELinux is important (and very useful) security for containers. A lot of its usual criticisms (hard to use, difficult to debug) are almost completely mitigated with containers, where we can use very restrictive default policies on containers (which should never talk to the host system outside of the limited ways we explicitly allow them to, like volume mounts). SELinux has stopped a few container CVEs before, including the recent `runc` vulnerability.

Unfortunately, when the container gains new default requirements, we have to race to get things updated to make sure they don't break, and it looks like we were too late here. There's a new containers-selinux that's being pushed into updates-testing soon (hopefully later today) which has a fix.

**Comment by lfarkas** [comment]
container-selinux-2.99-1.git9a53d6c.fc29 don't solve the problem.

but anyway why podman, libpod, runc or anything else depend on selinux??? selinux by design should have to be an outer tool the whole container infrastructure shouldn't have to be know anything about selinux!

**Comment by mheon** [comment]
We need to know a fair bit about SELinux to make it work with containers. Each container has a unique MCS label, which is automatically allocated to prevent duplication (to prevent inter-container breakouts). We need to generate those labels and use them to label the container's filesystem and the container process, when it starts.

In this case, though, the problem isn't in Podman's SELinux implementation, but the default system SELinux policy for containers, which is restricting a syscall made by `runc`.

We're waiting on `container-selinux-2.100.1.git3b78187.fc29` right now, which is still pending push to the repos.

**Comment by lfarkas** [comment]
who restricting runc? selinux is disabled on my system! 
container-selinux only influence the selinux behaviour but if I disable it than it shouldn't have to be any effect.
anyway does it means that on a system which disable selinux podman and containers can't be used?

btw: container-selinux-2.100-1.git3b78187.fc29 do NOT solve the problem.

**Comment by mheon** [comment]
@rhatdan PTAL - .100 doesn't seem to have fixed the SELinux disabled issues with keycreate

**Comment by rhatdan** [comment]
This is a runc issue, downgrade the version of runc, or wait grab the one I am building in koji right now.  WIll put in updates testing ASAP

**Comment by rhatdan** [comment]
runc-1.0.0-92.dev.gitc1b8c57.fc29.src.rpm

**Comment by rhatdan** [comment]
Could you grab the package from https://bodhi.fedoraproject.org/updates/FEDORA-2019-6174b47003 and test it out. If it fixes the issue, could you update the Karma.

**Comment by lfarkas** [comment]
yes

**Comment by giuseppe** [comment]
@lfarkas can this be closed now?

**Comment by krystalcode** [comment]
Is there a solution currently proposed? This is still happening in F34. The package mentioned above is for F29.

**Comment by rhatdan** [comment]
Please open a new issue, if you are seeing,even if it is an old issue.  Most likely what you are seeing is different then the original reported issue.
